Tarea clase 15 - Correciones

jjuannn opened the pull request:
Agrega correciones para la tarea de la clase 15

jjuannn committed:
actualiza

jjuannn committed:
Merge branch 'tarea-clase-13' of https://github.com/jjuannn/pokedex into tarea-clase-13

jjuannn committed:
actualiza

jjuannn committed:
modifica api para uso de entidades

las paginas se guardan en otro modulo

jjuannn committed:
agrega funciones para entidades

agrega funciones para usar clases

jjuannn committed:
modifica las funciones existentes

modifica las funciones ya existentes para poder usar clases

jjuannn committed:
arregla cantidad de pokemones

no mostraba la cantidad de pokemones porque estaba mal modificada la funcion

jjuannn committed:
Revert "modifica api para uso de entidades"

This reverts commit 5f94aa52d26e0fb1bcfc9ca7ac2762f4ea66302c.

jjuannn committed:
Revert "agrega funciones para entidades"

This reverts commit b3f69b89792740796a55e72f819110599929bc69.

jjuannn committed:
Revert "Revert "modifica api para uso de entidades""

This reverts commit df33bf45885dc0c899da36cfc3a11c76a89d44e6.

jjuannn committed:
Revert "Revert "agrega funciones para entidades""

This reverts commit c3c6629dc4dc13b4956a44559e369559dbe68623.

jjuannn committed:
Update src/servicios/servicios.js

Co-authored-by: Fabricio Leonardo Sodano Pascazi <fabricio.sodano@gmail.com>

jjuannn committed:
agrega url guardada en una const

agrega la const baseURL y las usa en los fetch para no escribir muchas veces la misma URL

jjuannn committed:
actualiza la clase pagina

actualiza la clase pagina, borra parametros que no le correspondian y los corrige

jjuannn committed:
actualiza la clase pokemon

actualiza la clase pokemon. Ahora podemos acceder a las estadisticas del pokemon directamente

jjuannn committed:
borra comentario

borra comentario

jjuannn committed:
actualiza mapeador

actualiza mapeador para aplicar las correciones de pokemon.js y pagina.js

jjuannn committed:
actualiza el modulo servicios

Borra lineas innecesarias y console.log. tambien borra la funcion que obtenia la pagina anterior desde api, ya que si cuando cargamos una pagina la guardamos en local storage, esta funcion era innecesaria

jjuannn committed:
actualiza nombre de las funciones

cambia los nombres de las funciones y deja de usar abreviaturas

jjuannn committed:
Arregla errores varios

cambia las constantes dentro de actualizarInformacion, para que la actualize con lo que le devuelve el mapeador.

fsodano commented on src/api/api.js:
```diff
@@ -1,51 +1,26 @@
-      return respuesta
+    .then(respuestaApi => {
+      const primeraPagina = nuevaPagina(respuestaApi)
+      guardarPagina("0", primeraPagina)
```
por qué la API sabe del servicio? El módulo de API se encarga de interactuar con la API de pokemon, y nada más.

fsodano commented on src/mapeador/mapeador.js:
```diff
@@ -0,0 +1,32 @@
+import {Pokemon} from "../entidades/pokemones.js"
+import {Pagina} from "../entidades/paginas.js"
+
+export function nuevoPokemon(respuesta){
+    const {
+        name: nombre,
+        height: altura,
+        types: tipos,
+        sprites: { front_default: foto },
+        weight: peso,
+        stats: estadisticas
+    } = respuesta
+
+    const hp = estadisticas[0].base_stat
+    const ataque = estadisticas[1].base_stat
+    const defensa = estadisticas[2].base_stat
+    const ataqueEspecial = estadisticas[3].base_stat
+    const defensaEspecial = estadisticas[4].base_stat
+    const velocidad = estadisticas[5].base_stat
+
+    return new Pokemon(nombre, altura, tipos, foto, peso, hp, ataque, defensa, ataqueEspecial, defensaEspecial, velocidad)
+}
```
excelente

fsodano commented on src/mapeador/mapeador.js:
```diff
@@ -0,0 +1,32 @@
+        count: totalPokemones,
+        next: proximaPagina,
+        previous: paginaAnterior,
+        results: listaPokemones
```
cada uno de estos pokemones los tenés que pasar por la función nuevoPokemon, porque listaPokemones debería ser un array de clases de tipo Pokemon.

Fijate de hacerlo con la función `map()`

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,38 @@
+
+export async function cargarPaginaSiguiente () {
+  try {
+    const paginaLS = buscarPaginaEnLS(offset)
```
`paginaCacheada` ó `paginaLocalStorage`, no abrevies!

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,38 @@
+    const respuestaApi = await obtenerPaginaSiguienteDesdeAPI(offset)
+    const proximaPagina = nuevaPagina(respuestaApi)
+    guardarPagina(offset, proximaPagina)
+    return proximaPagina
```
fijate cómo implementeste esto en `cargarPokemon` que lo hacés con 1 solo return al final, lo mismo deberías hacer en esta función!

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,38 @@
+  return pokemon
+}
+export async function cargarPaginaAnterior () {
+    const paginaLS = buscarPaginaEnLocalStorage(offset)
+    return paginaLS
```
```suggestion
    return buscarPaginaEnLocalStorage(offset)
```

fsodano commented on src/ui/ui.js:
```diff
@@ -1,27 +1,29 @@
+export async function crearListaPokemones(pokemones) {
   const $listaPokemones = document.querySelector('#lista-pokemones')
-    pokemones.forEach(pokemon => {
+  // PREGUNTAR SI listadoPokemones ESTA BIEN
+  const listadoPokemones = pokemones.listaPokemones
+    listadoPokemones.forEach(pokemon => {
```
Creo que la variable está de más, no la usás en ningún lado.
```suggestion
    pokemones.listaPokemones.forEach(pokemon => {
```

fsodano commented on src/ui/ui.js:
```diff
@@ -1,27 +1,29 @@
 import { buscarPokemonEnApi, obtenerInfoPokemones } from '../api/api.js'
+
 export async function inicializar(){
   const infoPokemones = await obtenerInfoPokemones()
```
esto está llamando a la API directamente, la UI no interactúa con la API ni con localStorage, sólo con el servicio.

fsodano replied:
y tampoco se tiene que llamar infoPokemones, en tu aplicación a esta entidad la estás llamando Pagina, así que debería ser `const pagina = obtenerPagina()` (del servicio)

jjuannn committed:
Update src/servicios/servicios.js

Co-authored-by: Fabricio Leonardo Sodano Pascazi <fabricio.sodano@gmail.com>

jjuannn committed:
Modifica funciones de servicios

Agrega funcion que obtiene la primer pagina desde API/LocalStorage para usarse en UI y que esta no se comunique directamente con la API

jjuannn committed:
Merge branch 'tarea-clase-13' of https://github.com/jjuannn/pokedex into tarea-clase-13

jjuannn committed:
actualiza UI

actualiza la funcion inicializar para que no se comunique directamente con api sino que lo haga con servicios.
Ademas modifica crearListaPokemones para poder usar los elementos de clase Pokemon

jjuannn committed:
actualiza mapeador

actualiza mapeador para crear elementos de clase pokemon cuando se cree una nueva pagina

jjuannn committed:
borra funciones innecesarias

borra lineas desactualizadas / que no se usaban

jjuannn committed:
actualiza parametros de clase Pagina

el parametro listaPokemones es un array, esto con la finalidad de crear elementos de clase pokemon.

jjuannn committed:
actualiza funciones de API

modifica la funcion obtenerInfoPokemones (renombrada) ya que estaba mal hecha

jjuannn committed:
actualiza la funcion nuevaPagina

jjuannn committed:
agrega atributos

agrega atributos para el body para arreglar un bug que al recargar la pagina  aveces se scrolleaba automaticamente hasta abajo de la misma, y al no tener la scrollbar, habia que cerrar y volver a abrir la pestaña para que se vea bien la pokedex

jjuannn committed:
elimina la funcion map

elimina la funcion map de la funcion nuevaPagina, ya que no era necesario que cada pokemon de la listaPokemones pase por la funcion nuevoPokemon.
Tambien vuelve la UI a como estaba antes para poder usar correctamente la informacion que devuelve la funcion nuevaPagina

jjuannn committed:
elimina node_modules del repo. remoto

No tenia que comitear node_modules
